crates/epaint/src/tessellator.rs

```diff
@@ -1315,6 +1318,74 @@ impl Tessellator {
             .stroke_closed(self.feathering, stroke, out);
     }
 
+    /// Tessellate a single [`EllipseShape`] into a [`Mesh`].
+    ///
+    /// * `shape`: the ellipse to tessellate.
+    /// * `out`: triangles are appended to this.
+    pub fn tessellate_ellipse(&mut self, shape: EllipseShape, out: &mut Mesh) {
+        let EllipseShape {
+            center,
+            radius,
+            fill,
+            stroke,
+        } = shape;
+
+        if radius.x <= 0.0 || radius.y <= 0.0 {
+            return;
+        }
+
+        if self.options.coarse_tessellation_culling
+            && !self
+                .clip_rect
+                .expand2(radius + Vec2::splat(stroke.width))
+                .contains(center)
+        {
+            return;
+        }
+
+        // Get the max pixel radius
+        let max_radius = radius.max_elem() * self.pixels_per_point;
+        // Get the power of two below the radius to limit the number of vertices
+        let floored_radius = 2_f32.powf(max_radius.log2().floor()) as u32;
```

Comment:
could use `next_power_of_two`

Reply:
why round to power-of-two though?

Reply:
Not sure why I chose rounding to a power of 2, have changed to just scale with the radius